parsers/ELEXON.py

```diff
@@ -124,337 +240,146 @@ def query_additional_eso_data(
-        axis=1,
+    merged_events = ProductionBreakdownList.merge_production_breakdowns(
+        [parsed_events_fuelhh, parsed_events_eso], logger
     )
-    df = df.rename(columns=ESO_FUEL_MAPPING)
-    return df.set_index("datetime")
+    return merged_events.to_list()
 
 
-def process_production_events(
-    fuel_inst_data: pd.DataFrame, eso_data: pd.DataFrame
-) -> list[dict]:
-    """Combine FUELINST report and ESO data together to get the full picture and to EM Format."""
-    df = fuel_inst_data.join(eso_data, rsuffix="_eso")
-    df = df.rename(columns={"wind_eso": "wind", "solar_eso": "solar"})
-    df = df.groupby(df.columns, axis=1).sum()
-    data_points = []
-    for time_t in pd.unique(df.index):
-        time_df = df[df.index == time_t]
-
-        data_point = {
-            "zoneKey": "GB",
-            "datetime": time_t.to_pydatetime(),
-            "source": "bmreports.com",
-            "production": {},
-            "storage": {},
+def query_exchange(
+    zone_key: ZoneKey, session: Session, target_datetime: datetime, logger: Logger
+) -> list:
+    all_exchanges: list[ExchangeList] = []
+    for interconnector in ZONEKEY_TO_INTERCONNECTOR[zone_key]:
+        exchange_params = {
+            "settlementDateFrom": (target_datetime - timedelta(days=2)).strftime(
+                "%Y-%m-%d"
+            ),
+            "settlementDateTo": target_datetime.strftime("%Y-%m-%d"),
+            "interconnectorName": interconnector,
+            "format": "json",
         }
-
-        for row in time_df.iterrows():
-            electricity_production = row[1].to_dict()
-            for key in electricity_production:
-                if key in PRODUCTION_MODES:
-                    data_point["production"][key] = electricity_production[key]
-                elif key == "hydro storage":
-                    # According to National Grid Eso:
-                    # The demand due to pumping at hydro pump storage units; the -ve signifies pumping load.
-                    # We store the pump loading as a positive value and discharge as negative.
-                    data_point["storage"]["hydro"] = -electricity_production[key]
-
-        data_points.append(data_point)
-    return data_points
-
-
-def parse_production(
-    csv_text: str,
-    target_datetime: datetime | None = None,
-    logger: Logger = getLogger(__name__),
-):
-    if not csv_text:
-        return None
-
-    report = REPORT_META["B1620"]
-
-    # create DataFrame from slice of CSV rows
-    df = pd.read_csv(StringIO(csv_text), skiprows=report["skiprows"] - 1)
-
-    # check field count in report is as expected
-    field_count = len(df.columns)
-    if field_count != report["expected_fields"]:
-        raise ValueError(
-            "Expected {} fields in B1620 report, got {}".format(
-                report["expected_fields"], len(df.columns)
+        exchange_data = query_elexon(
+            ELEXON_URLS["exchange"], session, exchange_params
+        ).get("data")
```

Comment:
There is no guarantee that get is available since the response is a JSON response, we need to ensure it is a dict before calling get on it.

The type data also says it will return a list, which causes type issues.

Reply:
This is just a type issue so ignoring it for now since we need to get this merged ASAP.